key/key.go

```diff
@@ -97,6 +109,16 @@ func (k Key256) HexString() string {
 	return hex.EncodeToString(k.b[:])
 }
 
+// MarshalBinary marshals the key into a byte slice.
+// The bytes may be passed to NewKey256 to construct a new key with the same value.
+func (k Key256) MarshalBinary() ([]byte, error) {
```

Comment:
You could consider an interface assertion along `var _ encoding.BinaryMarshaler = Key256{}` and `var _ encoding.BinaryUnmarshaler = Key256{}`

Reply:
Then I would have to import the encoding package simply for a test assertion. Could do it in key_test.go